### PR TITLE
Warning when authz passed with an error

### PR DESCRIPTION
**Pull request by yue9944882**
**What this PR does / why we need it**:
As the title
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go [hunk 1]
```diff
@@ -61,6 +61,9 @@ func WithAuthorization(handler http.Handler, a authorizer.Authorizer, s runtime.
 		if authorized == authorizer.DecisionAllow {
 			audit.LogAnnotation(ae, decisionAnnotationKey, decisionAllow)
 			audit.LogAnnotation(ae, reasonAnnotationKey, reason)
+			if err != nil {
+				glog.Warningf("authorization %v passed but with an error: %v", req.RequestURI, err.Error())
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go]: no need for `.Error()`

Comment (neolit123) [reply staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go]: `"authorization for %q passed but with an error: %v"`
seems better.

File: staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go [hunk 2]
```diff
@@ -61,6 +61,9 @@ func WithAuthorization(handler http.Handler, a authorizer.Authorizer, s runtime.
 		if authorized == authorizer.DecisionAllow {
 			audit.LogAnnotation(ae, decisionAnnotationKey, decisionAllow)
 			audit.LogAnnotation(ae, reasonAnnotationKey, reason)
+			if err != nil {
+				glog.Warningf("authorization for %q passed but with an error: %v", req.RequestURI, err)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go]: is this actionable or especially concerning? if not, I'm not sure we should flood the logs with this.

Comment (yue9944882) [reply staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go]: @liggitt I think we should at least output sth when a decision is make with an error. Or we won't have any idea of underlying errors from authn?

Comment (neolit123) [reply staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go]: you can also use `glog.V(N)....` here where `N` is a verbosity level.

Comment (krmayankk) [reply staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go]: Interesting , how does authorization pass with errors :-) ? Just curious what errors are those ?

**Pull request closed without merging**